app/controllers/admin/api/service_contracts_controller.rb

```diff
@@ -37,9 +31,7 @@ def destroy
   # Service Subscription Update
   # PUT /admin/api/accounts/{account_id}/service_contracts/{id}.xml
   def update
-    service = service_contract.issuer
-    new_plan = service.service_plans.find(service_contract_plan_id)
-    service_contract.change_plan!(new_plan)
+    service_contract.change_plan!(service_plan)
```

Comment:
We already have `#service_plan` that fetches the service plan by ID, we are reusing it here.
We are further validating that the new plan belongs to the same service as the current plan, by `validate_same_service` hook.

Reply:
agreed